x86/proofs/x86.ml

```diff
@@ -1417,6 +1417,17 @@ let x86_VPSLLQ = new_definition
         let res:(128)word = usimd2 (\z. word_shl z count) (word_zx x) in
         (dest := (word_zx res):N word) s`;;
 
+let x86_VPERMQ = new_definition
+  `x86_VPERMQ dest src imm8 (s:x86state) =
+      let (x:N word) = read src s
+      and od = read imm8 s in
+      if dimindex(:N) = 256 then
+        let res:(256)word = usimd4 (\(od:(2)word).
+            (word_subword:int256->num#num->int64) (word_zx x) ((val od)*64,64)) od in
+        (dest := (word_zx res):N word) s
+      else
+        raise_exception x86_Exception_UD s`;;
```

Comment:
Is this the correct way to handle this? 

VPERM2I128 is an AVX2-only instruction that operates exclusively on 256-bit YMM registers. According to the Intel Software Developer's Manual, VPERM2I128 has no 128-bit (XMM) variant - it only exists in the 256-bit form. When the instruction is attempted on 128-bit operands, the processor generates an Undefined Instruction (#UD) exception.